lib/name.mli

```diff
@@ -27,11 +27,20 @@ open Cstruct
 type label
 
 (** Domain name, as a list of strings. *)
-type domain_name = string list
+type domain_name
```

Comment:
i'd prefer to do this change after this PR, otherwise it will pollute the diff files, making them difficult to read